addons/google_drive/models/google_drive.py

```diff
@@ -0,0 +1,185 @@
+        headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+        data_json = json.dumps({"title": name_gdocs, "description": record_url, "parents": parents_dict['parents']})
+        req = urllib2.Request(request_url, data_json, headers)
+        content = json.loads(urllib2.urlopen(req, timeout=TIMEOUT).read())
+        res = {}
+        if content.get('alternateLink'):
+            attach_vals = {'res_model': res_model, 'name': name_gdocs, 'res_id': res_id, 'type': 'url', 'url': content['alternateLink']}
+            res['id'] = self.env['ir.attachment'].create(attach_vals).id
+            # Commit in order to attach the document to the current object instance, even if the permissions has not been written.
+            self.env.cr.commit()
+            res['url'] = content['alternateLink']
+            key = self._get_key_from_url(res['url'])
+            request_url = "https://www.googleapis.com/drive/v2/files/%s/permissions?emailMessage=This+is+a+drive+file+created+by+Odoo&sendNotificationEmails=false&access_token=%s" % (key, access_token)
+            try:
+                req = urllib2.Request(request_url, json.dumps({'role': 'writer', 'type': 'anyone', 'value': '', 'withLink': True}), headers)
+                urllib2.urlopen(req, timeout=TIMEOUT)
+            except urllib2.HTTPError:
+                raise self.env['res.config.settings'].get_config_warning(_("The permission 'reader' for 'anyone with the link' has not been written on the document"))
+            if self.env.user.email:
+                try:
+                    req = urllib2.Request(request_url, json.dumps({'role': 'writer', 'type': 'user', 'value': self.env.user.email}), headers)
+                    urllib2.urlopen(req, timeout=TIMEOUT)
+                except urllib2.HTTPError:
+                    pass
+        return res
+
+    @api.model
+    def get_google_drive_config(self, res_model, res_id):
+        '''
+        Function called by the js, when no google doc are yet associated with a record, with the aim to create one. It
+        will first seek for a google.docs.config associated with the model `res_model` to find out what's the template
+        of google doc to copy (this is usefull if you want to start with a non-empty document, a type or a name
+        different than the default values). If no config is associated with the `res_model`, then a blank text document
+        with a default name is created.
+          :param res_model: the object for which the google doc is created
+          :return: the config id and config name
+        '''
+        if not res_id:
+            raise UserError(_("Creating google drive may only be done by one at a time."))
+        # check if a model is configured with a template
+        configs = []
+        for config in self.search([('model_id', '=', res_model)]):
+            if config.filter_id:
+                if (config.filter_id.user_id and config.filter_id.user_id.id != self.env.uid):
```

Comment:
@msh-odoo, Here we needs first condition.
If we omit first condition and ```user_id``` is not set (means ```False```) then condition will be
```
if False != self.env.uid:  # result True and continue
    continue
```
So, here comparing ```self.env.uid``` only when ```user_id``` is not ```False```.